src/main/scala/shine/OpenCL/CodeGenerator.scala

```diff
@@ -183,10 +184,10 @@ class CodeGenerator(override val decls: CCodeGenerator.Declarations,
 
     case IdxVec(_, _, i, e) => CCodeGen.codeGenIdx(i, e, env, path, cont)
 
-    case OpenCLFunction(name, _, _, args) =>
+    case OpenCLFunctionCall(name, _, _, args) =>
```

Comment:
should this be `ocl.FunctionCall`?

Reply:
It is a call to built in OpenCL functions (so it's not a "function call in OpenCL" but rather a call to an "OpenCL function").
That's why I kept the name as it should not just be reflected by the package its in.

Does that make sense?

Reply:
I see, I think that I have personally always been using regular C function call primitives even for OpenCL built-in functions such as `dot`.